### PR TITLE
wide option for advanced table block

### PR DESCRIPTION
**Pull request by danielfmiranda**
## Description

**Link to test page:** https://foundation-s-10163-wide-fbv9zi.mofostaging.net/en/publication-page-with-child-article-pages/fixed-title-article-page/

**Username/Password:** admin2/admin2

**Related PRs/issues:** #10163

This PR adds a new checkbox/boolean field named "wide" to the article page's `AdvancedTableBlock`.

When this field is **checked**, the table extends beyond the other content of the article page. 
When this field is **unchecked**, the table is in line with the other content.

This field is set to `True` by default. 

## Screenshots:

**"wide" set to True**
Screenshot 2023-03-15 at 09-40-09 Opaque and Overstretched Part II

**"wide" set to False**
Screenshot 2023-03-15 at 09-40-35 Opaque and Overstretched Part II

### CODE REVIEW

File: network-api/networkapi/wagtailpages/migrations/0079_advance_table_block_wide_option.py
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/migrations/0079_advance_table_block_wide_option.py:1]: This is just the auto-generated migration file needed to add the "wide" field to the `AdvancedTableBlock` model.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/advanced_table_block.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {% load l10n wagtailcore_tags %}
 
 {% block block_content %}
-    <div class="article-table-block table-responsive">
+    <div class="article-table-block tw-overflow-x-auto tw-mx-auto {% if self.wide %} tw-w-full {% else %} tw-w-4/5 {% endif %}">
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/advanced_table_block.html:5]: The bootstrap class `table-responsive` added the following CSS Rules:

```css
overflow-x: auto;
width: 100%;
```

I am adding the same rules, this time using tailwind classes.

If the table is not set to "wide" then the width is reduced to 80% to match the width of the other article content.

**Comment by beccaklam** [comment]
@danielfmiranda Desktop and mobile look good but on tablet the "wide" looks a more narrow than the rest of the content, would we be able to make it so that it's the same width as the other streamfield block content? 
image

**Comment by mmmavis** [comment]
@beccaklam when I tested it the result seems to be different than what you described. 🤔 

On medium size screen, wide version seems to be in the same width of rest of the content while the regular version is narrower than other content. 

image

**Comment by mmmavis** [comment]
@danielfmiranda is there a mockup we can reference for tablet design?

**Comment by danielfmiranda** [comment]
Thanks for taking a look @beccaklam! I have added a fix for the error you found.  

Hi @mmmavis! I actually do not have a mockup. I believe that is the expected behavior, but I will leave it to Rebecca to confirm 👍

**Comment by beccaklam** [comment]
@mmmavis I think we had the same result? Wide set to false was resulting in a narrower width, while wide set to true was resulting in a width that matched the rest of the content. 

@danielfmiranda The fix looks good! Thank you for doing that :)

Yeah I'm not sure what the intended behaviour is for wide on tablet and mobile but I think at those screen widths having everything match in width is probably the best since there's not a lot of space already.

**Review by beccaklam — APPROVED** [review]

**Review by mmmavis — APPROVED** [review]
Nice work!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/advanced_table_block.html (hunk 1 above)
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/advanced_table_block.html]: Smart!

**Pull request merged**